design-principles.md

```diff
@@ -25,6 +25,10 @@
 1. [Avoid implementing templating logic](https://docs.google.com/document/d/1h_3vSApIsuiwGkrqSiegi4NVaYG4oVzBquGAhIN6qGM/edit#heading=h.6kxvcvm7rs3r); prefer variable replacement.
 1. Avoid implementing our own expression syntax; when required prefer existing languages which are widely used and include supporting development tools.
 1. In TEPs, discuss how the proposal affects the flexibility of Tekton and demonstrate that any specific/opinionated choices are necessary but extensible. 
+1. Before adding an optional field to the spec or status of `CustomRun`, consider
```

Comment:
This feels very specific to Pipelines. Perhaps somewhere in the pipelines documentation is a better place for this?

Maybe https://github.com/tektoncd/pipeline/tree/main/docs/developers?

Reply:
I agree, Triggers' ClusterInterceptor is also similar in that we except external users to implement instead of just in-tree implementations.

Reply:
Thanks for the feedback! I was trying to think of a more general way to express this since I know we have multiple api extension points in tekton projects but I think it'll be too confusing/generic-- I'll open a PR for this in pipelines instead.